### PR TITLE
NAS-107370 / 20.10 / Make sure hook is being called for a specific pool

### PR DESCRIPTION
**Pull request by william-gr**

### CODE REVIEW

**Comment by william-gr** [comment]
retest

**Comment by william-gr** [comment]
retest this please

**Comment by themylogin** [comment]
@william-gr shouldn't we query all pools and sync zfs guid for each then?

**Comment by william-gr** [comment]
@themylogin No, I am just fixing a traceback, I am not even convinced that hook is required to begin with, sysevent.fs.zfs.config_sync is already called when a pool is imported.

**Comment by themylogin** [comment]
@william-gr then why don't you remove the hook completely?

**Comment by william-gr** [comment]
@themylogin because I am not 100% sure thats the case and I didnt create the plugin.

**Comment by themylogin** [comment]
@william-gr let me re-check and remove it then?

**Comment by william-gr** [comment]
As you wish

**Comment by themylogin** [comment]
@william-gr another important fix we want in 12.0. I have not found other renamed ZFS events like this.

File: src/middlewared/middlewared/plugins/disk_/zfs_guid.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ async def sync_zfs_guid(self, pool_id_or_pool):
 
 async def devd_zfs_hook(middleware, data):
     if data.get("type") in (
-        "misc.fs.zfs.config_sync",
+        "sysevent.fs.zfs.config_sync",
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/disk_/zfs_guid.py:52]: Did this event name was actually renamed in devd? Did you verify yourself?

Comment (themylogin) [reply src/middlewared/middlewared/plugins/disk_/zfs_guid.py:52]: `sysevent.fs.zfs.config_sync` is the event that is now appearing in devd pipe if that's what you're talking about, I have not been examining actual ZFS code.

**Pull request merged**